### DISCUSSION TITLE
Obfuscation vs Masquerade

### DISCUSSION THREAD

**Discussion by hossein1376** [apernet/hysteria#1115] [Q&A]
I have always used obfs with little to no issues. Recently, I was going through the documentation and saw this bit: 
`If your network specifically blocks QUIC or HTTP/3 traffic (but not UDP in general), obfuscation can be used to work around this.`

Also:
`NOTE: Enabling obfuscation will make your server incompatible with standard QUIC connections and it will no longer function as a valid HTTP/3 server.`

QUIC or HTTP/3 is not blocked on my network and I can connect to Hysteria server with masquerade option enabled. But which method recommend and is more resistant against possible detection or blocking? (I live in Iran for context)

**Comment by meower1** [comment]
As specified in the documentation the obfuscation option makes the traffic seem like random udp packets, while hysteria's main functionality is hiding itself as plain http/3 traffic so its best to avoid using obfuscation unless you cant connect without it. as for masquerade it would be dependent on the firewall situation and we can't say for sure wether its effective or not but having it on would probably be best

**Comment by shakibamoshiri** [comment]
Hy2 default connection protocol is h3, if did not work yet UDP is allow go with obfuscation
Masquerading is for hiding your end application ( a fake app like an h3 web server )

it is better to use default h3 + masquerade
But if h3 (plain) did not connect, h3 + obfs 

NOTE
If we enable masquerade and obfs , the connection protocol to connect to that end app is not longer h3, but h2 . It means if you open your web browser while obfs has been enabled ,  the browser does not understand obfs and falls back to h2

**Comment by ParsaJR** [comment]
> But which method recommend and is more resistant against possible detection or blocking?

Hi. 

I've been using Hysteria2 in Iran with the obfuscation method, and it has been working great for about 8 months, with a traffic usage of about 500 gigabytes per month. 



Honestly, I haven't tested the 'Masquerade' method, but it might be fine too.
